
float32 avg_error
float32 max_error
